dataset-workflows/simple_s3_ingest/cdk/cdk_stack.py

```diff
@@ -0,0 +1,180 @@
+import os
+import config
+from aws_cdk import core, aws_iam, custom_resources
+import aws_cdk.aws_stepfunctions as stepfunctions
+import aws_cdk.aws_events as events
+import aws_cdk.aws_events_targets as targets
+from aws_cdk import aws_secretsmanager as secretsmanager
+from aws_cdk import aws_lambda
+from aws_cdk import aws_stepfunctions_tasks as tasks
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_sqs as sqs
+from aws_cdk import aws_s3 as s3
+
+
+class CdkStack(core.Stack):
+    def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+        stack_name = construct_id
+
+        bucket = "climatedashboard-data"
+        name = "simple-s3-ingest"
+
+        s3bucket = s3.Bucket.from_bucket_name(
+            self, f"{id}-bucket", bucket_name=bucket
+        )
+
+        ec2_network_access = aws_iam.PolicyStatement(
+            actions=[
+                "ec2:CreateNetworkInterface",
+                "ec2:DescribeNetworkInterfaces",
+                "ec2:DeleteNetworkInterface",
+            ],
+            resources=["*"],
+        )
+        full_bucket_access = aws_iam.PolicyStatement(
+            actions=["s3:GetObject", "s3:PutObject"],
+            resources=[f"arn:aws:s3:::{bucket}/*"],
+        )
+
+        database_vpc = ec2.Vpc.from_lookup(self, f"{id}-vpc", vpc_id=config.VPC_ID)
+
+        database_security_group = ec2.SecurityGroup.from_security_group_id(self, "SG", config.SECURITY_GROUP_ID)
+
+        lambda_function_security_group = ec2.SecurityGroup(
+            self,
+            f"{id}-lambda-sg",
+            vpc=database_vpc,
+            description="fromCloudOptimizedPipelineLambdas",
+        )
+
+        lambda_function_security_group.add_egress_rule(
+            ec2.Peer.any_ipv4(),
+            connection=ec2.Port(protocol=ec2.Protocol("ALL"), string_representation=""),
+            description="Allow lambda security group all outbound access",
+        )
+
+        database_security_group.add_ingress_rule(
+            lambda_function_security_group,
+            connection=ec2.Port.tcp(5432),
+            description="Allow pgstac database write access to lambda",
+        )
+
+        # Discover function
+        s3_discovery_lambda = aws_lambda.Function(
+            self,
+            f"{id}-discover-fn",
+            code=aws_lambda.Code.from_asset_image(
+                directory="../../lambdas/s3-discovery",
+                file="Dockerfile",
+                entrypoint=["/usr/local/bin/python", "-m", "awslambdaric"],
+                cmd=["handler.handler"],
+            ),
+            handler=aws_lambda.Handler.FROM_IMAGE,
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=1024,
+            timeout=core.Duration.seconds(30),
+        )
```

Comment:
Instead of using a Lambda container image, you may be able to get better performance and simplicity by leaning on CDK's [PythonFunction construct](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-lambda-python-alpha.PythonFunction.html) to bundle the handler as a standard Python lambda.

Reply:
Good call. Will check.